pkg/planner/cascades/rule/rule_type.go

```diff
@@ -14,13 +14,18 @@
 
 package rule
 
+import "github.com/pingcap/tidb/pkg/planner/cascades/memo"
+
 type ruleType int
 
 const (
-	// DefaultNone indicates this is none rule.
-	DefaultNone ruleType = iota
+	// DEFAULT_NONE indicates this is none rule.
+	DEFAULT_NONE ruleType = iota
```

Comment:
Oh, we just use this `DEFAULT_NONE` as the ID of the BaseRule? or make BaseRule's as `intest.Assert(false, "should not be called")`

Reply:
yes, make sense